### ISSUE TITLE
.unique() to have a callback ?

### ISSUE THREAD

**Issue by code2be** [rschmukler/agenda#413]
Hello,

I'm using .unique() to make sure the same job doesn't get created more than once at the same time.
But, I feel the need of having a callback, which will be simply called once the duplicate job get rejected because of .unique().

So, What do You think about adding such callback ? Is it existing somewhere and I don't know ? Somebody worked on it so I don't do it again myself ?

Thank You.

**Comment by michelem09** [comment]
The callback is the `save()` function you have to call after creating/modifying the job.

```
job.unique({ 'something': true });
job.save(function (err) {
   if (err) console.log(err);
});
```

**Comment by emhagman** [comment]
@michelem09 Does this callback tell you whether there was a unique constraint issue? I believe that is what @code2be is referring to.

**Comment by michelem09** [comment]
Yes it should. Because it calls a `findOneAndUpdate` and return the errors/results.

https://github.com/agenda/agenda/blob/b818de4d47a3182b95db327edb12f861738957b0/lib/agenda.js#L362
